compiler/rustc_codegen_rmc/src/codegen/span.rs

```diff
@@ -27,6 +27,16 @@ impl<'tcx> GotocCtx<'tcx> {
         )
     }
 
+    // Get the location of the caller. This will attempt to reach the macro caller.
+    pub fn codegen_caller_span(&self, sp: &Option<Span>) -> Location {
+        if let Some(span) = sp {
+            let topmost = span.ctxt().outer_expn().expansion_cause().unwrap_or(*span);
```

Comment:
what is this doing?

Reply:
It is getting the location of the macro caller. It gets the outer most expansion and get its cause.

Reply:
I think the important part is the expansion_cause() method. This is the doc:
```
    /// Returns span for the macro which originally caused this expansion to happen.
    ///
    /// Stops backtracing at include! boundary.
    pub fn expansion_cause(mut self) -> Option<Span> 
```

This is how other backend engines assemble the Location in PanicInfo when the current function does not have `#[track_caller]` attribute which we currently don't support.

Reply:
I agree this function could use an extensive doc comment. It looks like you found a really great solution compared to our original hack, so helping us use this function more in the future would be awesome.